p2p/protocol/identify/id.go

```diff
@@ -445,10 +445,7 @@ func newStreamAndNegotiate(ctx context.Context, c network.Conn, proto protocol.I
 		log.Debugw("error opening identify stream", "peer", c.RemotePeer(), "error", err)
 		return nil, err
 	}
-	err = s.SetDeadline(time.Now().Add(Timeout))
-	if err != nil {
-		return nil, err
-	}
```

Comment:
This seems wrong. Shouldn't mock networks support deadlines rather than us silently timing out? 

Also a comment for why we are ignoring the error would be helpful.